apps/io-wallet-web-func/src/wallet-instance.ts

```diff
@@ -0,0 +1,71 @@
+import { IsoDateFromString } from "@pagopa/ts-commons/lib/dates";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as O from "fp-ts/Option";
+import * as RTE from "fp-ts/ReaderTaskEither";
+import * as TE from "fp-ts/TaskEither";
+import { pipe } from "fp-ts/function";
+import * as t from "io-ts";
+import { EntityNotFoundError } from "io-wallet-common/http-response";
+import { JwkPublicKey } from "io-wallet-common/jwk";
+import { User } from "io-wallet-common/user";
+
+const WalletInstanceValid = t.type({
+  createdAt: IsoDateFromString,
+  hardwareKey: JwkPublicKey,
+  id: NonEmptyString,
+  isRevoked: t.literal(false),
+  signCount: t.number,
+  userId: User.props.id,
+});
+
+type WalletInstanceValid = t.TypeOf<typeof WalletInstanceValid>;
+
+const WalletInstanceRevoked = t.type({
+  createdAt: IsoDateFromString,
+  hardwareKey: JwkPublicKey,
+  id: NonEmptyString,
+  isRevoked: t.literal(true),
+  revokedAt: IsoDateFromString,
+  signCount: t.number,
+  userId: User.props.id,
+});
```

Comment:
You can use a generic type and then specialize, for example:

```suggestion

const WalletInstanceBase = t.type({
  createdAt: IsoDateFromString,
  hardwareKey: JwkPublicKey,
  id: NonEmptyString,
  signCount: t.number,
  userId: User.props.id,
});


const WalletInstanceValid = t.intersection([
  WalletInstanceBase,
  t.type({
    isRevoked: t.literal(false)
  })
])

type WalletInstanceValid = t.TypeOf<typeof WalletInstanceValid>;

const WalletInstanceRevoked = t.intersection([
  WalletInstanceBase,
  t.type({
    isRevoked: t.literal(true),
    revokedAt: IsoDateFromString,
  })
])
```